components/style/properties/helpers/animated_properties.mako.rs

```diff
@@ -1362,6 +1364,88 @@ impl Animatable for MozLength {
     }
 }
 
+impl Animatable for SVGStrokeLength {
+    #[inline]
+    fn add_weighted(&self, other: &Self, self_portion: f64, other_portion: f64) -> Result<Self, ()> {
+        match (self.0, other.0) {
+            (NumberOrPercentage::Number(from),
+             NumberOrPercentage::Number(to)) => {
+                Ok(SVGStrokeLength(NumberOrPercentage::Number(
+                    try!(from.add_weighted(&to, self_portion, other_portion)))))
+            },
+            (NumberOrPercentage::Percentage(from),
+             NumberOrPercentage::Percentage(to)) => {
+                Ok(SVGStrokeLength(NumberOrPercentage::Percentage(
+                    try!(from.add_weighted(&to, self_portion, other_portion)))))
+            },
```

Comment:
So, other functions handle interpolations of numbers and percentages using `Calc`. We should probably do the same. Could you also mention it from here?